client/mm/libxc/binance.go

```diff
@@ -1915,7 +1915,38 @@ func binanceMarketToDexMarkets(binanceBaseSymbol, binanceQuoteSymbol string, tok
 	return markets
 }
 
+type BNApiResponse struct {
+	Code int    `json:"code"`
+	Msg  string `json:"msg"`
+}
+
 func requestInto(req *http.Request, thing interface{}) error {
-	// bnc.log.Tracef("Sending request: %+v", req)
-	return dexnet.Do(req, thing, dexnet.WithSizeLimit(1<<24))
+	var sizeLimit int64 = 1 << 24
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error performing request: %w", err)
+	}
+	defer resp.Body.Close()
+	// Binance returns 400 Bad Request for errors
+	if resp.StatusCode == http.StatusBadRequest {
```

Comment:
I don't think `400` is the only error response code we could see that might inlcude more info in the response body.
![image](https://github.com/user-attachments/assets/4dbb8697-44ec-46ff-a3a0-1a1b3b47a444)

Reply:
400 was the only one I could reproduce.  Do you think it's OK to change this to `if resp.StatusCode != http.StatusOK` ?

Reply:
> 400 was the only one I could reproduce. Do you think it's OK to change this to `if resp.StatusCode != http.StatusOK` ?

There may be different codes in the 200s that represent success. You should check `if resp.StatusCode >= 200 && resp.StatusCode <= 299`

Reply:
As per the docs, 200 is the only status being returned for success:
https://binance-docs.github.io/apidocs/websocket_api/en/#status-codes